### PR TITLE
*: support flashback truncate table.

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
This is for PR: https://github.com/pingcap/tidb/pull/13109

Add syntax :
```sql
flashback table t until timestamp '2019-11-04 11:56:49.497 +0800 CST' to t1; 
```

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - Possible performance regression

Related changes

### CODE REVIEW

File: parser.y [hunk 1]
```diff
@@ -4482,7 +4522,7 @@ UnReservedKeyword:
 | "WITHOUT" | "RTREE" | "EXCHANGE" | "COLUMN_FORMAT" | "REPAIR" | "IMPORT" | "DISCARD" | "TABLE_CHECKSUM" | "UNICODE"
 | "SQL_TSI_DAY" | "SQL_TSI_HOUR" | "SQL_TSI_MINUTE" | "SQL_TSI_MONTH" | "SQL_TSI_QUARTER" | "SQL_TSI_SECOND" |
 "SQL_TSI_WEEK" | "SQL_TSI_YEAR" | "INVISIBLE" | "VISIBLE" | "TYPE" | "NOWAIT" | "REPLICA" | "LOCATION" | "LABELS"
-| "LOGS" | "HOSTS" | "AGAINST" | "EXPANSION"
+| "LOGS" | "HOSTS" | "AGAINST" | "EXPANSION" | "FLASHBACK" | "UNTIL"
```
Comment (tiancaiamao) [inline parser.y]: 1. `FLASHBACK` belongs to NotKeywordToken
2. Don't forget to add `UNTIL` to `TestSimple()` in parser_test.go

Ref: https://dev.mysql.com/doc/refman/8.0/en/keywords.html

**Comment by kennytm** [comment]
Is the syntax finalized? Wanna note that [Oracle's FLASHBACK syntax](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/FLASHBACK-TABLE.html) is slightly different.

```sql
FLASHBACK TABLE t TO TIMESTAMP '2019-10-31 23:59:58';
FLASHBACK TABLE t TO BEFORE DROP RENAME TO `t1`;
FLASHBACK DATABASE db TO TIMESTAMP '2019-10-31 23:59:58';
FLASHBACK DATABASE db TO BEFORE TIMESTAMP '2019-10-31 23:59:59';
```

**Comment by crazycs520** [comment]
@kennytm, The syntax is finalized, the difference with oracle is intentional

**Comment by tiancaiamao** [comment]
LGTM @kennytm

**Pull request merged**